apis/core/v1beta1/openstackversion_types.go

```diff
@@ -204,3 +207,42 @@ func init() {
 func (instance OpenStackVersion) IsReady() bool {
 	return instance.Status.Conditions.IsTrue(condition.ReadyCondition)
 }
+
+func getOpenStackReleaseVersion(openstackReleaseVersion string, releaseVersionScheme string, operatorConditionName string) string {
```

Comment:
Can we add a test for this function to bring some clarity to the intended behaviour?

I think I get it, but if we have a functional test to validate that regex along with a couple of valid and invalid examples, I think it would be helpful.

Reply:
good call, added a few test cases